ouroboros-consensus/src/Ouroboros/Consensus/Node/Tracers.hs

```diff
@@ -0,0 +1,96 @@
+{-# LANGUAGE FlexibleContexts #-}
+module Ouroboros.Consensus.Node.Tracers
+  ( -- * All tracers of a node bundled together
+    Tracers' (..)
+  , Tracers
+  , nullTracers
+  , showTracers
+    -- * Specific tracers
+  , TraceForgeEvent (..)
+  ) where
+
+import           Control.Tracer (Tracer, nullTracer, showTracing)
+
+import           Ouroboros.Network.Block (HasHeader, Point, SlotNo)
+import           Ouroboros.Network.BlockFetch (FetchDecision,
+                     TraceFetchClientState, TraceLabelPeer)
+import           Ouroboros.Network.TxSubmission.Inbound
+                     (TraceTxSubmissionInbound)
+import           Ouroboros.Network.TxSubmission.Outbound
+                     (TraceTxSubmissionOutbound)
+
+import           Ouroboros.Consensus.Block (Header)
+import           Ouroboros.Consensus.BlockFetchServer
+                     (TraceBlockFetchServerEvent)
+import           Ouroboros.Consensus.ChainSyncClient (TraceChainSyncClientEvent)
+import           Ouroboros.Consensus.ChainSyncServer (TraceChainSyncServerEvent)
+import           Ouroboros.Consensus.Mempool.API (GenTx, GenTxId,
+                     TraceEventMempool)
+import           Ouroboros.Consensus.TxSubmission
+                     (TraceLocalTxSubmissionServerEvent (..))
+
+{-------------------------------------------------------------------------------
+  All tracers of a node bundled together
+-------------------------------------------------------------------------------}
+
+data Tracers' peer blk f = Tracers
+  { chainSyncClientTracer         :: f (TraceChainSyncClientEvent blk)
+  , chainSyncServerTracer         :: f (TraceChainSyncServerEvent blk)
+  , blockFetchDecisionTracer      :: f [TraceLabelPeer peer (FetchDecision [Point (Header blk)])]
+  , blockFetchClientTracer        :: f (TraceLabelPeer peer (TraceFetchClientState (Header blk)))
+  , blockFetchServerTracer        :: f (TraceBlockFetchServerEvent blk)
+  , txInboundTracer               :: f (TraceTxSubmissionInbound  (GenTxId blk) (GenTx blk))
+  , txOutboundTracer              :: f (TraceTxSubmissionOutbound (GenTxId blk) (GenTx blk))
+  , localTxSubmissionServerTracer :: f (TraceLocalTxSubmissionServerEvent blk)
+  , mempoolTracer                 :: f (TraceEventMempool blk)
+  , forgeTracer                   :: f (TraceForgeEvent blk)
+  }
+
+-- | A record of 'Tracer's for the node.
+type Tracers m peer blk = Tracers' peer blk (Tracer m)
```

Comment:
Why did this need to be defined in two parts? When do we need these things to be not just a `Tracer` ?

Reply:
To avoid having to define `TraceOptions` like in https://github.com/input-output-hk/cardano-node/pull/82/files#diff-95c531b5b63fb72bec464a1f996906d6R58. Instead of having to keep these in-sync (which will annoying seeing that these repos often drift out of sync), you can just get `TraceOptions` like so:
```haskell
type TraceOptions = Tracers' () () (Const Bool)
```
and then:

```haskell
parseTraceOptions :: Parser TraceOptions
parseTraceOptions = Tracers
    <$> (Const <$> parseTraceChainSyncClient)
    <*> (Const <$> parseTraceChainSyncServer)
    <*> (Const <$> parseTraceBlockFetchDecision)
    <*> (Const <$> parseTraceBlockFetchClient)
    <*> (Const <$> parseTraceBlockFetchServer)
    <*> (Const <$> parseTraceTxInbound)
    <*> (Const <$> parseTraceTxOutbound)
    <*> (Const <$> parseTraceLocalTxSubmissionServer)
    <*> (Const <$> parseTraceMempool)
    <*> (Const <$> parseTraceForge)
```

Reply:
:thinking: I was thinking to do the same in `cardano-node` but there the set of options is smaller than the set of tracers.

Reply:
We should talk to @CodiePP about the config for this. I don't think we're expecting to have a whole bunch of bool options, but rather to wire this up to the switchboard, and then they have more generic config for turning on/off/details of tracers.

I think we can use types to ensure these things do not drift without needing to over-paramaterise our tracers bundle type.